ouroboros-consensus/test/consensus-test/Test/Consensus/MiniProtocol/ObjectDiffusion/PerasCert/Smoke.hs

```diff
@@ -0,0 +1,133 @@
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE UndecidableInstances #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
+{-# OPTIONS_GHC -Wno-unused-top-binds #-}
+
+module Test.Consensus.MiniProtocol.ObjectDiffusion.PerasCert.Smoke (tests) where
+
+import Control.Monad.IOSim (IOSim)
+import Control.Tracer (Tracer (..), contramap, nullTracer)
+import Data.Functor.Identity (Identity (..))
+import qualified Data.List.NonEmpty as NE
+import Debug.Trace (traceM)
+import Network.TypedProtocol.Driver.Simple (runPeer, runPipelinedPeer)
+import Ouroboros.Consensus.Block.SupportsPeras
+import Ouroboros.Consensus.MiniProtocol.ObjectDiffusion.ObjectPool.API
+import Ouroboros.Consensus.MiniProtocol.ObjectDiffusion.ObjectPool.PerasCert
+import Ouroboros.Consensus.Storage.PerasCertDB.API
+  ( AddPerasCertResult (..)
+  , PerasCertDB
+  , PerasCertTicketNo
+  )
+import qualified Ouroboros.Consensus.Storage.PerasCertDB.API as PerasCertDB
+import qualified Ouroboros.Consensus.Storage.PerasCertDB.Impl as PerasCertDB
+import Ouroboros.Consensus.Util.IOLike
+import Ouroboros.Network.Block (Point (..), SlotNo (SlotNo), StandardHash)
+import Ouroboros.Network.Point (Block (Block), WithOrigin (..))
+import Ouroboros.Network.Protocol.ObjectDiffusion.Codec
+import Ouroboros.Network.Protocol.ObjectDiffusion.Inbound
+  ( objectDiffusionInboundServerPeerPipelined
+  )
+import Ouroboros.Network.Protocol.ObjectDiffusion.Outbound (objectDiffusionOutboundClientPeer)
+import Test.Consensus.MiniProtocol.ObjectDiffusion.Smoke
+  ( ListWithUniqueIds (..)
+  , WithId
+  , getId
+  , prop_smoke_object_diffusion
+  )
+import Test.QuickCheck
+import Test.Tasty
+import Test.Tasty.QuickCheck (testProperty)
+import Test.Util.TestBlock
+
+-- When wanting to debug a part of the test:
+debugTracer :: Monad m => Tracer m String
+debugTracer = Tracer traceM
```

Comment:
FYI this is already available by default from [`contra-tracer`](https://github.com/input-output-hk/iohk-monitoring-framework/blob/1c8a7704003d453c8633f290a8101248a9874b70/contra-tracer/src/Control/Tracer.lhs#L259-L260)

Also, especially when shrinkers come up, we usually use [`recordingTracerTVar`](https://ouroboros-consensus.cardano.intersectmbo.org/haddocks/ouroboros-consensus-unstable-consensus-testlib/Test-Util-Tracer.html#v:recordingTracerTVar) (or one of its variants) instead (and print the traces in the `counterexample` output), as this avoids cluttering the output in the happy path.